core/src/main/java/org/apache/druid/metadata/EnvironmentVariableDynamicConfigProvider.java

```diff
@@ -22,21 +22,31 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.collect.ImmutableMap;
+import org.apache.druid.java.util.common.logger.Logger;
 
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Objects;
 
-public class EnvironmentVariableDynamicConfigProvider implements DynamicConfigProvider
+public class EnvironmentVariableDynamicConfigProvider implements DynamicConfigProvider<String>
 {
-  private final ImmutableMap<String, String> variables;
+
+  private static final Logger log = new Logger(EnvironmentVariableDynamicConfigProvider.class);
+
+  private ImmutableMap<String, String> variables;
 
   @JsonCreator
   public EnvironmentVariableDynamicConfigProvider(
       @JsonProperty("variables") Map<String, String> config
   )
   {
-    this.variables = ImmutableMap.copyOf(config);
+    try {
+      this.variables = ImmutableMap.copyOf(config);
+    }
+    catch (NullPointerException e) {
+      log.error(e, "Can not parse config by EnvironmentVariableDynamicConfigProvider! Please check your config file.");
+      throw e;
+    }
```

Comment:
It would be nice to propagate the error to users so that they don't have to look up logs when this happens.

```suggestion
    this.variables = ImmutableMap.copyOf(Preconditions.checkNotNull(config, "config"));
```

Reply:
fixed